testsuite/features/step_definitions/command_steps.rb

```diff
@@ -1563,3 +1563,29 @@
   raise 'File injection failed' unless return_code.zero?
   node.run('salt-call --local --file-root=/usr/share/susemanager/salt --module-dirs=/usr/share/susemanager/salt/ --log-level=info --retcode-passthrough state.apply ' + state)
 end
+
+And(/^I upload "([^\"]*)" to "([^\"]*)" on "([^\"]*)"$/) do |file, targetdir, host|
```

Comment:
Steps calling steps are a pain. Just nuke this whole step, and call directly `file_inject()` below :smile_cat: .

Reply:
I agree with @Bischoff , but if by chance we keep this step definition to upload a file to a "target path" into a host, we should do it in a way that is homogeneous with this other step:
 https://github.com/uyuni-project/uyuni/blob/master/testsuite/features/step_definitions/command_steps.rb#L425-L429
```
When(/^I copy "([^"]*)" to "([^"]*)"$/) do |file, host|
  node = get_target(host)
  return_code = file_inject(node, file, File.basename(file))
  raise 'File injection failed' unless return_code.zero?
end
````

So, I would propose something like this:
```suggestion
And(/^I copy "([^\"]*)" to "([^\"]*)" inside this path "([^\"]*)"$/) do |file, host, targetdir|
```